libs/constellation/src/constellation.cpp

```diff
@@ -569,9 +560,9 @@ bool Constellation::Run(UriSet const &initial_peers, core::WeakRunnable bootstra
     reactor_.Start();
 
     /// BLOCK EXECUTION & MINING
-
     execution_manager_->Start();
     tx_processor_.Start();
+    main_chain_service_->Start();
```

Comment:
To avoid a race before setting the genesis file, the gossip handler for blocks is attached after genesis file load. Unfortunetly the RPC calls can still go through.